pkgs/applications/misc/diary/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, ncurses, readline }:
+
+stdenv.mkDerivation rec {
+  version = "0.3";
+  name = "diary-${version}";
+
+  src = fetchFromGitHub {
+    owner = "in0rdr";
+    repo = "diary";
+    rev = "v${version}";
+    sha256 = "0sda8hrnhs7zvl10iri6srllcimi719yvzn22g9jmiryi9sllay6";
+  };
+
+  buildInputs = [ ncurses readline ];
+
+  installPhase = ''
```

Comment:
Please move the creation of the directories into `preInstalledPhase` and pass `PREFIX=$out` via `installFlags`.